subcommands/keys/tuf_rotate_offline_key.go

```diff
@@ -0,0 +1,380 @@
+package keys
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	canonical "github.com/docker/go/canonical/json"
+	tuf "github.com/theupdateframework/notary/tuf/data"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+)
+
+const (
+	tufCmdAnnotation          = "tuf-cmd"
+	tufCmdRotateOfflineKey    = "rotate-offline-key"
+	tufCmdRotateRootLegacy    = "rotate-root"
+	tufCmdRotateTargetsLegacy = "rotate-targets"
+)
+
+func init() {
+	rotate := &cobra.Command{
+		Use:   "rotate-offline-key --role root|targets --creds=<offline-creds.tgz>",
+		Short: "Rotate the offline TUF signing key for the Factory",
+		Long: `Rotate the TUF root or TUF targets offline signing key for the Factory.
+
+The new signing key is rotated in both CI and production TUF root transactionally.
+
+When you rotate the TUF targets offline signing key:
+- if there are any production targets in your factory, they are re-signed using the new key.
+- if there is an active wave in your factory, the TUF targets rotation is not allowed.`,
+		Run:         doRotateOfflineKey,
+		Annotations: map[string]string{tufCmdAnnotation: tufCmdRotateOfflineKey},
```

Comment:
@mike-sul I'm sorry for this change in the middle of your review; the use of annotations helped to simplify the Run attribute usage quite some.